src/prisma/generator/schema.py

```diff
@@ -80,7 +80,8 @@ class Schema(BaseModel):
 
     @classmethod
     def from_data(cls, data: AnyData) -> 'Schema':
-        models = [Model(info=model) for model in data.dmmf.datamodel.models]
+        active_provider = data.datasources[0].active_provider
+        models = [Model(info=model, active_provider=active_provider) for model in data.dmmf.datamodel.models]
```

Comment:
nit: I think plumbing the provider through like this is a little awkward, we have a similar problem in other places and the solution I came up with is to set a [`ContextVar`](https://github.com/RobertCraigie/prisma-client-py/blob/64e0ddc6f137c57960930446f727160bc1cf3788/src/prisma/generator/models.py#L123). I don't think that solution is *amazing* either but I think it is better than having to pass through information like this.
```py
from .models import data_ctx

# ...

if data_ctx.get().datasources[0].active_provider: ...
```